File: app/src/main/java/io/eodc/planit/helper/AssignmentInfoInputHelper.java
package io.eodc.planit.helper;

import android.support.design.widget.TextInputLayout;
import android.text.Editable;
import android.text.TextWatcher;

public class AssignmentInfoInputHelper implements TextWatcher {
    private TextInputLayout mLayoutEditTitle;

    public AssignmentInfoInputHelper(TextInputLayout layoutEditTitle) {
        this.mLayoutEditTitle = layoutEditTitle;
    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {

    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {

        if (s.toString().equals("")) {
            mLayoutEditTitle.setError("Title can't be empty");
        } else {
            mLayoutEditTitle.setError("");
        }
    }

    @Override
    public void afterTextChanged(Editable s) {

    }


}

File: app/src/main/java/io/eodc/planit/helper/AssignmentInfoInputHelper.kt
package io.eodc.planit.helper

import com.google.android.material.textfield.TextInputLayout
import android.text.Editable
import android.text.TextWatcher

class AssignmentInfoInputHelper(private val mLayoutEditTitle: TextInputLayout) : TextWatcher {

    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}

    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

        if (s.toString() == "") {
            mLayoutEditTitle.error = "Title can't be empty"
        } else {
            mLayoutEditTitle.error = ""
        }
    }

    override fun afterTextChanged(s: Editable) {}
}
